procrastinate/worker.py

```diff
@@ -341,30 +386,15 @@ async def run(self):
                             queues=self.queues,
                         ),
                     )
-                    return
-
-                while True:
-                    await self._wait_for_job()
+                    self._stop_event.set()
+
+                while not self._stop_event.is_set():
+                    # wait for a new job notification, a stop even or the next polling interval
+                    await utils.wait_any(
```

Comment:
Hm, I would have used `wait_for` for the timeout but it's probably roughly equivalent to use sleep. I don't know if one has an benefit compared to the other, but this one is certainly readable. The other one would need to add a `try/except TimeoutError`, just to `pass` it, so, nice.